src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpPreferDictionaryContainsMethods.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.NetCore.Analyzers.Runtime;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Runtime
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpPreferDictionaryContainsMethods : PreferDictionaryContainsMethods
+    {
+        private protected override bool TryGetPropertyReferenceOperation(IInvocationOperation containsInvocation, [NotNullWhen(true)] out IPropertySymbol? propertySymbol)
```

Comment:
Seems we have a very similar extension method in the repo: https://github.com/dotnet/roslyn-analyzers/blob/8cddd3dfe9b67a6c4e6868f94fc095537ee1f833/src/Utilities/Compiler/Extensions/IOperationExtensions.cs#L22-L27

Do you think we can re-use it for the first part of this method that computes the receiver?

Reply:
Ah that one is to get the receiver type, not the operation.